File: ws/first_ws/src/topic_pkg/src/custom_msg_publisher.cpp
//
// Created by zcb on 2021/4/2.
//

#include <iostream>
#include <ros/ros.h>
#include <demo_msgs/Student.h>

using namespace std;

int main(int argc,char** argv){
    string nodeName = "publisher_custom_msg_cpp_node";
    ros::init(argc,argv,nodeName);
    ros::NodeHandle node;

    string topicName = "/custom/msg/topic";
    ros::Publisher pub = node.advertise<demo_msgs::Student>(topicName,1000);

    ros::Rate rate(1);
    demo_msgs::Student stu;
    int idx = 0;
    while(ros::ok()){
        stu.name = "tom"+to_string(idx++);
        stu.age = idx;
        pub.publish(stu);
        rate.sleep();
    }

    ros::spin();

    return 0;
}

File: 06ros/ws01/first_ws/src/topic_pkg/scripts/custom_msg_publisher.py
#! /usr/bin/env python
# coding:utf8

import rospy
from demo_msgs.msg import Student

if __name__ == '__main__':
    nodeName = "publish_custom_msg_py_node"
    rospy.init_node(nodeName)

    topicName = "/custom/msg/topic"
    pub = rospy.Publisher(topicName, Student, queue_size=1000)

    rate = rospy.Rate(1)
    stu = Student()
    idx = 0
    while not rospy.is_shutdown():
        stu.name = "tom %d" % idx
        stu.age = idx
        pub.publish(stu)
        idx += 1
        rate.sleep()

    rospy.spin()
